homeassistant/components/discord/notify.py

```diff
@@ -64,10 +68,16 @@ async def async_send_message(self, message, **kwargs):
         embeds: list[nextcord.Embed] = []
         if ATTR_EMBED in data:
             embedding = data[ATTR_EMBED]
+            title = embedding.get(ATTR_EMBED_TITLE) or nextcord.Embed.Empty
+            description = embedding.get(ATTR_EMBED_DESCRIPTION) or nextcord.Embed.Empty
+            color = embedding.get(ATTR_EMBED_COLOR) or nextcord.Embed.Empty
+            url = embedding.get(ATTR_EMBED_URL) or nextcord.Embed.Empty
             fields = embedding.get(ATTR_EMBED_FIELDS) or []
 
             if embedding:
-                embed = nextcord.Embed(**embedding)
+                embed = nextcord.Embed(
```

Comment:
I see there is also `Embed.from_dict` which looks like it may do exactly what we want?
https://nextcord.readthedocs.io/en/latest/api.html#nextcord.Embed.from_dict
and reading the implementation it seems to do exactly what this function is doing:
https://github.com/nextcord/nextcord/blob/408862e24c11bf7d7a95a56d2eb2532094845404/nextcord/embeds.py#L209

What do you think?

Reply:
Yea, I was looking at this as well. Functionally, from_dict works. My only concern is that there are other keys that also work in the dict that are not documented in the home assistant documentation, which would be more of a feature-add. This would mean any time the nextcord library is upgraded and introduces more fields, the documentation would need to be updated as well.

For the scope of this PR, I just wanted to align back to the previous working behavior.

Reply:
Agree that we don't want to be chasing documentation.  It seems like we should be able to either limit to fields specified in this file, or update the docs to point to the nextcoord docs, or not worry about it.

Just to be clear i am not thrilled with how verbose this is given it can be replaced with a single line. I am also not sure that any documentation needs to be updated at all.

Reply:
That's fair. It is rather verbose.

The new keynames that would be accepted are 'video', 'provider', and 'image'. (Not sure how the image key is different from the one that exists outside of Embed). These are not documented by home assistant, and would need to be filtered out until those features are tested.

We could use from_dict and provide a dictionary that has been filtered with an allow-list of tested features. That would also work to reduce all of the embed.set_* calls.

Is this something that you want accomplished in this PR? My rationale was just to make the smallest change possible to unblock the integration from working.

Reply:
If I were doing this I would just add the from dict call and I wouldn't worry about the docs personally. 

My second choice would add it with a one line filter by the set of allowed keys already referenced here.

I don't consider either of these very big. Though the smallest change word be to rollback the broken move to nextcoord since it didn't really work.

Reply:
I didn't intend for this to be a big deal as I thought it would be fast to switch to. I'll merge this and we can deal with it in a follow up.

Reply:
Hi apologies, it's not a big deal! I just don't have spare cycles at the moment to keep working on it. I can revisit this in a week or so and look into the additional features as well.